devito/operator/operator.py

```diff
@@ -168,6 +169,15 @@ def _build(cls, expressions, **kwargs):
         # Python-level (i.e., compile time) and C-level (i.e., run time) performance
         profiler = create_profile('timers')
 
+        # Add missing implicit dimensions in the presence of a Break equation
+        if any(isinstance(i, Break) for i in expressions):
+            for e in expressions:
+                dims = list(e.implicit_dims)
+                for expr in preorder_traversal(e):
```

Comment:
How large is `e`? We tried to get rid of `preorder_traversal` everywhere a while back because it's quite heavy and slow.

Reply:
Removed `preorder_traversal` in favor of a proper devito.symbolics function `retrieve_dimensions`.